src/Chapter1.hs

```diff
@@ -554,7 +559,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int 
+mid x y z
+  | (x >= y && x <= z) || (x <= y && x >= z) = x
+  | (y >= x && y <= z) || (y <= x && y >= z) = y
+  | (z >= x && z <= y) || (z <= x && z >= y) = z
+  | otherwise = x
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

But in this case, you can use `otherwise` instead of the third case, as we sure that when we come here, the only possible answer is `z`:

```suggestion
  | otherwise = z
```

Reply:
You're totally right!!